### PR TITLE
Simplify configuration class methods

### PR DESCRIPTION
**Pull request by davidglezz**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Simplify some methods in Configuration class.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | Running current `Configuration` class tests.
| Possible impacts? | no

### CODE REVIEW

File: classes/Configuration.php [hunk 1]
```diff
@@ -657,25 +651,16 @@ public static function deleteById(int $configurationId): void
...
-        }
-
-        if ($context == Shop::CONTEXT_SHOP && Configuration::hasKey($key, $idLang, null, $idShop)) {
-            return true;
-        } elseif ($context == Shop::CONTEXT_GROUP && Configuration::hasKey($key, $idLang, $idShopGroup)) {
-            return true;
-        } elseif ($context == Shop::CONTEXT_ALL && Configuration::hasKey($key, $idLang)) {
-            return true;
+        if ($context != Shop::CONTEXT_SHOP && $context != Shop::CONTEXT_GROUP && $context != Shop::CONTEXT_ALL) {
+            return false;
```
Comment (davidglezz) [inline classes/Configuration.php]: If this case is impossible, we can remove these lines.
Update: Removed.

**Review by FabienPapet — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]
OK for me. Don't know if we need a QA approve, I let you decide.

**Comment by davidglezz** [comment]
I think this PR does not need QA

**Review by florine2623 — APPROVED** [review]
Hello @davidglezz ,

Launched [automated tests](https://github.com/florine2623/testing_pr/actions/runs/3548323742).
They're OK !

So it is QA ✅ !

**Comment by marsaldev** [comment]
Thanks @davidglezz ! 💪

**Pull request merged**